pilot/cmd/pilot-agent/main.go

```diff
@@ -166,6 +165,23 @@ var (
 		RunE: func(c *cobra.Command, args []string) error {
 			cmd.PrintFlags(c.Flags())
 
+			// must read proxy config first, we set environment vars here that are used elsewhere
+			proxyConfig, err := constructProxyConfig()
+			if err != nil {
+				return fmt.Errorf("failed to get proxy config: %v", err)
+			}
+			for k, v := range proxyConfig.ProxyMetadata {
```

Comment:
Does it mean we no longer need to inject all the env variables, just the proxy metadata ?